src/components/TrialBanner.js

```diff
@@ -61,10 +66,13 @@ export const FreeCreditsModal= ajaxCaller(class FreeCreditsModal extends Compone
   }
 
   render() {
-    const { onDismiss } = this.props
+    const { isActive } = this.props
     const { pageTwo, termsAgreed, cloudTermsAgreed, loading } = this.state
-    return h(Modal, {
-      onDismiss,
+    return isActive && h(Modal, {
```

Comment:
The typical pattern is to control the visibility of the modal from outside. That way the dismiss logic can be centralized in the `onDismiss` callback, and the component will unmount when dismissed, so the state will automatically reset.